content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/create_vm_import_request.rb

```diff
@@ -0,0 +1,46 @@
+#
+# Description: This method initiates VM import to given infra provider
+#
+
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module Import
+            class CreateVmImportRequest
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                options = {
```

Comment:
@matobet - Could you add basic validation that the required arguments exist before calling execute to create the automation request?

Here's a sample basic validation method:
def validate_root_args(arg_names)
  arg_names.each do |name|
    unless $evm.root[name].present?
      msg = "Error, required root attribute: #{name} not found"
      $evm.log(:error, msg)
      raise msg
    end
  end
end

and you can call it prior to calling execute....

validate_root_args(%w(vm dialog_name dialog_provider dialog_cluster dialog_sparse))